steppy/base.py

```diff
@@ -569,6 +654,15 @@ def _build_structure_dict(self, structure_dict):
             structure_dict['edges'].add((input_data, self.name))
         return structure_dict
 
+    def _register_step(self):
+        pass
+
+    def _set_mode(self, mode):
```

Comment:
I could potentially _set_mode('inference') and run transform on the train set in which case all the cache could be useful and so now that I think about it, I am not sure about the clean_cache_upstream